src/main/java/WordFrequencyGame.java

```diff
@@ -1,75 +1,40 @@
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.StringJoiner;
-import java.io.CharArrayWriter;
+import java.util.*;
 
-import java.time.LocalDateTime;
+import java.util.stream.Collectors;
 
 public class WordFrequencyGame {
-    public String getResult(String inputStr){
+    public String getResult(String inputStr) {
 
+        return inputStr.split("\\s+").length == 1 ? inputStr + " 1" : getCalculatedString(inputStr);
+    }
 
-        if (inputStr.split("\\s+").length==1) {
-            return inputStr + " 1";
-        } else {
-
-            try {
-
-                //split the input string with 1 to n pieces of spaces
-                String[] arr = inputStr.split("\\s+");
-
-                List<Input> inputList = new ArrayList<>();
-                for (String s : arr) {
-                    Input input = new Input(s, 1);
-                    inputList.add(input);
-                }
-
-                //get the map for the next step of sizing the same word
-                Map<String, List<Input>> map =getListMap(inputList);
-
-                List<Input> list = new ArrayList<>();
-                for (Map.Entry<String, List<Input>> entry : map.entrySet()){
-                    Input input = new Input(entry.getKey(), entry.getValue().size());
-                    list.add(input);
-                }
-                inputList = list;
-
-                inputList.sort((w1, w2) -> w2.getWordCount() - w1.getWordCount());
+    private String getCalculatedString(String inputStr) {
+        try {
 
-                StringJoiner joiner = new StringJoiner("\n");
-                for (Input w : inputList) {
-                    String s = w.getValue() + " " +w.getWordCount();
-                    joiner.add(s);
-                }
-                return joiner.toString();
-            } catch (Exception e) {
+            return getStrListMap(Arrays
+                    .stream(inputStr.split("\\s+"))
+                    .map(str -> new Input(str, 1))
+                    .collect(Collectors.toList()))
```

Comment:
can have getStrListMap as a stream instead of wrapping it around a portion of the code
as the end of ) will be hard to trace